### ISSUE TITLE
Inside the model : Multilabel classification loss function 

### ISSUE THREAD

**Issue by YBK75** [ThilinaRajapakse/simpletransformers#661]
Hi Thilina,

First of all thank you for this amazing work, you package arised my curiosity for nlp. 

My Question is : what algorithm adaptation (multi-class) have you made to perform a Multi-label classification ?

I believe that adapting neural network algorithm to classify multi-label instances requires three key steps:
- Creating a new error function that captures the characteristics of multi-label learning.
- Modify the network to minimize this new error function.
- Using threshold function to determine an output is in the relevant set of labels.

Can you provide more information about your approach ? 

Best regards & much respect from Paris ;)

**Comment by ThilinaRajapakse** [comment]
I am using [BCEWithLogitsLoss](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html) as the error function (instead of CrossEntropy as with binary/multiclass classification). This is a fairly straightforward replacement in the `forward()` function of the Transformer models.

https://github.com/ThilinaRajapakse/simpletransformers/blob/6a6ae049242d0c9cff9e597fc3639e4d5e7e63e2/simpletransformers/custom_models/models.py#L71-L74

There is no modification to the loss minimization. It uses the same optimizer as the other models.

A sigmoid function is used on the logits to get the probabilities of each label.

https://github.com/ThilinaRajapakse/simpletransformers/blob/6a6ae049242d0c9cff9e597fc3639e4d5e7e63e2/simpletransformers/classification/classification_model.py#L1111-L1111

Finally, a simple thresholding function is used to get binary values for each label.

https://github.com/ThilinaRajapakse/simpletransformers/blob/6a6ae049242d0c9cff9e597fc3639e4d5e7e63e2/simpletransformers/classification/classification_model.py#L1152-L1159
